Database/macros/TOD_network_prep.py

```diff
@@ -0,0 +1,264 @@
+#change to correct scenario
+change_primary_scenario(scen)
+
+#clear out extra link and turn attributes
+#  -- link attributes to clear
+extra_atts_link = [
+    ['@vauto' , f'class 1: SOV volume iter_{glob_iter}'],
+    ['@vsov1' , f'class 1: SOV VOT1 volume iter_{glob_iter}'],
+    ['@vsov2' , f'class 2: SOV VOT2 volume iter_{glob_iter}'],
+    ['@vsov3' , f'class 3: SOV VOT3 volume iter_{glob_iter}'],
+    ['@vhov2' , f'class 4: HOV2 volume iter_{glob_iter}'],
+    ['@vhov3' , f'class 4: HOV3+ volume iter_{glob_iter}'],
+    ['@vbplt' , f'class 5: b plate truck volume iter_{glob_iter}'],
+    ['@vlght' , f'class 5: light truck volume iter_{glob_iter}'],
+    ['@vmed' , f'class 6: medium truck volume iter_{glob_iter}'],
+    ['@vhevy' , f'class 7: heavy truck volume iter_{glob_iter}'],
+]
+
+#  -- turn attributes to clear
+extra_atts_turn = [
+    ['@tauto' , f'class 1: SOV volume iter_{glob_iter}'],
+    ['@tsov1' , f'class 1: SOV VOT1 volume iter_{glob_iter}'],
+    ['@tsov2' , f'class 2: SOV VOT2 volume iter_{glob_iter}'],
+    ['@tsov3' , f'class 3: SOV VOT3 volume iter_{glob_iter}'],
+    ['@thov2' , f'class 4: HOV2 volume iter_{glob_iter}'],
+    ['@thov3' , f'class 4: HOV3+ volume iter_{glob_iter}'],
+    ['@tbplt' , f'class 5: b plate truck volume iter_{glob_iter}'],
+    ['@tlght' , f'class 5: light truck volume iter_{glob_iter}'],
+    ['@tmed' , f'class 6: medium truck volume iter_{glob_iter}'],
+    ['@thevy' , f'class 7: heavy truck volume iter_{glob_iter}'],
+]
+
+#  -- overwrite link attributes with 0
+for att in extra_atts_link:
+    create_extra(
+        extra_attribute_type = "LINK",
+        extra_attribute_name = att[0],
+        extra_attribute_description = att[1],
+        extra_attribute_default_value = 0.0,
+        overwrite = True,
+        scenario = scen
+    )
+
+#  -- overwrite turn attributes with 0
+for att in extra_atts_turn:
+    create_extra(
+        extra_attribute_type = "TURN",
+        extra_attribute_name = att[0],
+        extra_attribute_description = att[1],
+        extra_attribute_default_value = 0.0,
+        overwrite = True,
+        scenario = scen
+    )
+
+#get hours per tod (but set period 1 == 5 hours)
+#format: {<time-of-day> : <number of hours>}
+
+tod_mult_dict = {
+    1:5,
+    2:1,
+    3:2,
+    4:1,
+    5:4,
+    6:2,
+    7:2,
+    8:2
+}
+
+tod_mult = tod_mult_dict[int(tod)] 
+
+
+#if global iter == 0, call two emme macros -- Ftime.Capacity and Arterial.Delay
+# INHERITED FROM MACRO:
+# ~# heither, 04-03-2014: run ftime.capacity and arterial.delay for iteration 0
+# ~# this ensures link attributes are updated if edits were made and the
+# ~# procedures were not run
+if int(glob_iter) == 0:
+    # print('  --calculating ftime_capacity and arterial_delay')
+    fc.link_capacity()              ## -- Calculate link capacity
+    ad.arterial_delay()             ## -- Calculate arterial delay
+    bus.busveq(tod, empFile, scen, scen_txt, tod_mult)  ## -- Calculate busveq

```

Comment:
```suggestion
    bus.busveq(tod, Path(modeller.desktop.project_file_name()).name, scen, scen_txt, tod_mult)  ## -- Calculate busveq
```
Without a file name parameter, get the file name from the Modeller object.

Reply:
Good.